hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -870,7 +961,12 @@ public void deleteFilesystem(TracingContext tracingContext)
       LOG.debug("deleteFilesystem for filesystem: {}",
               client.getFileSystem());
 
-      final AbfsRestOperation op = client.deleteFilesystem(tracingContext);
+      final AbfsRestOperation op;
+      if (getPrefixMode() == PrefixMode.BLOB) {
```

Comment:
Why would we not consider the useBlobEndpoint variable while making the decision here ?

Reply:
Because deleteFileSystem is not a HDFS API. This function is never called in any flow that comes from AzureBlobFileSystem. This is used only in test classes.

So, I thought it does not make sense to have fallback for this method as it will never be used.
Same is the case for other file system APIs: getFileSystemProperties() and setFileSystemProperties()